Admin/MenuNodeAdmin.php

```diff
@@ -28,14 +42,87 @@ protected function configureFormFields(FormMapper $formMapper)
     {
         $formMapper
             ->with('form.group_general')
-                ->add(
-                    'parent',
-                    'doctrine_phpcr_odm_tree',
-                    array('root_node' => $this->menuRoot, 'choice_list' => array(), 'select_root_node' => true)
-                )
+                ->add('parent', 'doctrine_phpcr_odm_tree', array(
+                    'root_node' => $this->menuRoot,
+                    'choice_list' => array(),
+                    'select_root_node' => true
+                ))
             ->end()
         ;
+
         parent::configureFormFields($formMapper);
+
+        if (null === $this->getParentFieldDescription()) {
+
+            // Add the choice for the node links "target"
+            $formMapper
+                ->with('form.group_general')
+                    ->add('linkType', 'choice_field_mask', array(
+                        'choices' => array_combine(
+                            $this->contentAwareFactory->getLinkTypes(),
+                            $this->contentAwareFactory->getLinkTypes()
+                        ),
+                        'map' => array(
+                            'route' => array('route'),
+                            'uri' => array('uri'),
+                            'content' => array('content', 'doctrine_phpcr_odm_tree'),
+                        ),
+                        'empty_value' => 'auto',
+                        'required' => false
+                    ))
+                    //->add('route', 'text', array('required' => false))
```

Comment:
See:

 > MenuNode admin does no longer show a "route" and "uri" field, but it shows one "link" field instead. The linkType select input will determine what the value of "link" is (if it's a route or uri)

And the form event listeners now.

Previously, it was like:

```
link type: [ auto | route | uri ]
route: [______]
uri: [______]
```

This means that in case of "route", the uri field needs to be left blank and in case of "uri", the route field needs to be left blank. In case of "auto", one of them should be blank.

I've now changed this to:

```
link type: [route | uri]
link: [_______]
```
In case link type is set to "route", the link field will be mapped to the route property of MenuNode. In case of "uri", the link field will be mapped tot the uri property of MenuNode.